sentry_ldap_auth/backend.py

```diff
@@ -56,7 +79,10 @@ def get_or_create_user(self, username, ldap_user):
         if not organizations or len(organizations) < 1:
             return model
 
-        member_role = getattr(settings, 'AUTH_LDAP_SENTRY_ORGANIZATION_ROLE_TYPE', 'member')
+        member_role = _get_effective_sentry_role(ldap_user.group_names)
+        if not member_role:
+            member_role = getattr(settings, 'AUTH_LDAP_SENTRY_ORGANIZATION_ROLE_TYPE', 'member')
```

Comment:
So if no role mapping matches, it'll automatically pick `AUTH_LDAP_SENTRY_ORGANIZATION_ROLE_TYPE` (or `member`)?

It does not sound right and might be a security hazard (especially since it's undocumented).
I am in the opinion that if `AUTH_LDAP_SENTRY_GROUP_ROLE_MAPPING` is non-empty, and the user isn't in any of the specified groups, it should be denied access.

Reply:
Seems fair, thank you. We have an additional security layer where it is used, I failed to take that into account.

Reply:
However, if `AUTH_LDAP_SENTRY_ORGANIZATION_ROLE_TYPE` is `none`, then it's kinda okay as Sentry will 500..

Reply:
Well actually, if an user is created with an invalid role, Sentry also crashes later when we try to delete it.. So we'd actually need a fix.

Reply:
I don't see the problem. The groups that are passed to `_get_effective_sentry_role(group_names)` is the set of groups that the user is in, according to the LDAP directory (`ldap_user.group_names`). If none of those groups are specified in the role mapping, it will default to `AUTH_LDAP_SENTRY_ORGANIZATION_ROLE_TYPE` (or `'member'`).

Reply:
Exactly, which means that every single one of your 1000 LDAP users will be
granted access to Sentry as members, regardless of whether their group was
part of the ‘member’ mapping or not. That effectively makes the ‘member’
mapping useless as well as the LDAP authorization, leaving it to being
authentication only (but the purpose of this PR is to bring some basic
authorization through the roles). Except if there is another way to
strictly limit which groups are whitelisted - but the only parameter that
looks like that seems to be taking a single DN.